sources/packages/backend/libs/integrations/src/esdc-integration/e-cert-integration/e-cert-file-handler.ts

```diff
@@ -99,97 +108,123 @@ export class ECertFileHandler extends ESDCFileHandler {
+      const sequenceGroup = `${sequenceGroupPrefix}_${getISODateOnlyString(
+        new Date(),
+      )}`;
+      let uploadResult: ECertUploadResult;
+      await this.sequenceService.consumeNextSequence(
+        sequenceGroup,
+        async (nextSequenceNumber: number, entityManager: EntityManager) => {
+          uploadResult = await this.processECert(
             nextSequenceNumber,
-          );
-
-          // Create the request filename with the file path for the e-Cert File.
-          const fileInfo = await this.createRequestFileName(
+            entityManager,
+            eCertIntegrationService,
+            offeringIntensity,
             fileCode,
-            nextSequenceNumber,
           );
+        },
+      );
+      return uploadResult;
+    } catch (error: unknown) {
+      if (
+        error instanceof CustomNamedError &&
+        error.name === ECERT_GENERATION_NO_RECORDS_AVAILABLE
+      ) {
+        return {
+          generatedFile: "none",
+          uploadedRecords: 0,
+        };
+      }
+      throw error;
+    }
+  }
 
-          // Creates the repository based on the entity manager that
-          // holds the transaction already created to manage the
-          // sequence number.
-          const disbursementScheduleRepo =
-            entityManager.getRepository(DisbursementSchedule);
-          await this.disbursementScheduleService.updateRecordsInSentFile(
-            disbursementIds,
-            now,
-            disbursementScheduleRepo,
-          );
+  /**
+   * Prepare the disbursements for e-Cert generation and upload the e-Cert file.
+   * @param sequenceNumber e-Cert sequence number.
+   * @param entityManager manages the current DB transaction.
+   * @param eCertIntegrationService Full-Time/Part-Time integration responsible
+   * for the respective integration.
+   * @param offeringIntensity disbursement offering intensity.
+   * @param fileCode file code applicable for Part-Time or Full-Time.
+   * @returns information of the uploaded e-Cert file.
+   * @throws CustomNamedError ECERT_GENERATION_NO_RECORDS_AVAILABLE
+   */
+  private async processECert(
+    sequenceNumber: number,
+    entityManager: EntityManager,
+    eCertIntegrationService: ECertIntegrationService,
+    offeringIntensity: OfferingIntensity,
+    fileCode: string,
+  ): Promise<ECertUploadResult> {
+    try {
+      const disbursements =
+        await this.eCertGenerationService.prepareDisbursementsForECertGeneration(
+          offeringIntensity,
+          entityManager,
+        );
+      if (!disbursements.length) {
+        throw new CustomNamedError(
+          `There are no records available to generate an e-Cert file for ${offeringIntensity}`,
+          ECERT_GENERATION_NO_RECORDS_AVAILABLE,
+        );
+      }
+      this.logger.log(
+        `Found ${disbursements.length} ${offeringIntensity} disbursements schedules.`,
+      );
+      const disbursementRecords = disbursements.map((disbursement) => {
```

Comment:
`const disbursementRecords = disbursements.map((disbursement) =>  this.createECertRecord(disbursement );`

Reply:
Changed.